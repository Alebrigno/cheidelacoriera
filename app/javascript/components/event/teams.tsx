import * as React from "react";
import { Game, Team, TransportMean } from "../../models";

interface EventTeamsProps {
  game: Game;
}

export class EventTeams extends React.Component<EventTeamsProps> {
  constructor(props: EventTeamsProps) {
    super(props);
  }

  render() {
    return (
      <div className="row">
        {this._formatTeam(this.props.game.teams.home)}
        {this._formatInfo(this.props.game)}
        {this._formatTeam(this.props.game.teams.away)}
      </div>
    );
  }

  _formatTeam = (team: Team) => {
    return (
      <div className="col-md-5 left event-team">
        <img alt={team.name} className="team-logo" src={team.imageUrl} />
        <p>{team.name}</p>
      </div>
    );
  };

  _formatInfo = (game: Game) => {
    const score = game.score;
    const transportMean = game.transportMean;
    const time = game.time;
    if (score) {
      return this._formatSimple(score);
    } else if (transportMean.iconClass) {
      return this._formatTransportMean(transportMean);
    } else {
      return this._formatSimple(time);
    }
  };

  _formatTransportMean = (transportMean: TransportMean) => {
    return (
      <div className="col-md-2 event-score">
        <i className={transportMean.iconClass} />
        <p>
          <small>{transportMean.name}</small>
        </p>
      </div>
    );
  };

  _formatSimple = (info: string) => {
    return <div className="col-md-2 event-score">{info}</div>;
  };
}
