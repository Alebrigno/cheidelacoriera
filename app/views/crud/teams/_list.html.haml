= crud_table(:name, :country, :url, :image_url, :description)
