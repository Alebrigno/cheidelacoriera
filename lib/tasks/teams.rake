# frozen_string_literal: true

namespace :teams do
  desc 'Serie A 2017/2018 Teams data'
  task seriea20172018: :environment do
    fp = Rails.root.join('data', 'teams_seriea_20172018.json')
    teams = JSON.parse(File.read(fp), symbolize_names: true).fetch(:teams)

    before_count = Team.count
    teams.each { |t| Team.create(**t) {} }
    after_count = Team.count

    puts "#{after_count - before_count} out of #{teams.count} teams created"
  end
end
