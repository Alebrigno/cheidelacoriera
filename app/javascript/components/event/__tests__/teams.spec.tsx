import { configure } from "enzyme";
import * as Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

import { EventTeams } from "../teams";
import { shallow } from "enzyme";
import * as React from "react";

describe("EventTeams", () => {
  const teams = {
    away: {
      imageUrl: "http://link-to-team-1",
      name: "Team 1",
    },
    home: {
      imageUrl: "http://link-to-team-2",
      name: "Team 2",
    },
  };
  const busTransportMean = {
    iconClass: "fa fa-bus",
    name: "bus",
  };

  const time = "2021-11-20";

  ["1-1", undefined].forEach((score) => {
    [busTransportMean, undefined].forEach((transportMean) => {
      test(`when score=${score}, transportMean=${JSON.stringify(
        transportMean
      )} and time=${time} `, () => {
        const game = {
          time: time,
          score: score,
          teams: teams,
          transportMean: transportMean,
        };
        const eventTeam = shallow(<EventTeams game={game} />);
        expect(eventTeam.instance().render()).toMatchSnapshot();
      });
    });
  });
});
