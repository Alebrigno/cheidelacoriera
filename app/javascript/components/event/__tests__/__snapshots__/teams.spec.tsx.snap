// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTeams when score=1-1, transportMean={"iconClass":"fa fa-bus","name":"bus"} and time=2021-11-20  1`] = `
<div
  className="row"
>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 2"
      className="team-logo"
      src="http://link-to-team-2"
    />
    <p>
      Team 2
    </p>
  </div>
  <div
    className="col-md-2 event-score"
  >
    1-1
  </div>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 1"
      className="team-logo"
      src="http://link-to-team-1"
    />
    <p>
      Team 1
    </p>
  </div>
</div>
`;

exports[`EventTeams when score=1-1, transportMean=undefined and time=2021-11-20  1`] = `
<div
  className="row"
>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 2"
      className="team-logo"
      src="http://link-to-team-2"
    />
    <p>
      Team 2
    </p>
  </div>
  <div
    className="col-md-2 event-score"
  >
    1-1
  </div>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 1"
      className="team-logo"
      src="http://link-to-team-1"
    />
    <p>
      Team 1
    </p>
  </div>
</div>
`;

exports[`EventTeams when score=undefined, transportMean={"iconClass":"fa fa-bus","name":"bus"} and time=2021-11-20  1`] = `
<div
  className="row"
>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 2"
      className="team-logo"
      src="http://link-to-team-2"
    />
    <p>
      Team 2
    </p>
  </div>
  <div
    className="col-md-2 event-score"
  >
    <i
      className="fa fa-bus"
    />
    <p>
      <small>
        bus
      </small>
    </p>
  </div>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 1"
      className="team-logo"
      src="http://link-to-team-1"
    />
    <p>
      Team 1
    </p>
  </div>
</div>
`;

exports[`EventTeams when score=undefined, transportMean=undefined and time=2021-11-20  1`] = `
<div
  className="row"
>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 2"
      className="team-logo"
      src="http://link-to-team-2"
    />
    <p>
      Team 2
    </p>
  </div>
  <div
    className="col-md-2 event-score"
  >
    2021-11-20
  </div>
  <div
    className="col-md-5 left event-team"
  >
    <img
      alt="Team 1"
      className="team-logo"
      src="http://link-to-team-1"
    />
    <p>
      Team 1
    </p>
  </div>
</div>
`;
